markets/perps-market/contracts/storage/LiquidationAssetManager.sol

```diff
@@ -0,0 +1,114 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.11 <0.9.0;
+
+import {ITokenModule} from "@synthetixio/core-modules/contracts/interfaces/ITokenModule.sol";
+import {IPerpRewardDistributor} from "@synthetixio/perps-reward-distributor/contracts/interfaces/IPerpsRewardDistributor.sol";
+import {ISynthetixSystem} from "../interfaces/external/ISynthetixSystem.sol";
+import {PerpsMarketFactory} from "./PerpsMarketFactory.sol";
+import {DecimalMath} from "@synthetixio/core-contracts/contracts/utils/DecimalMath.sol";
+
+/**
+ * @title LiquidationAssetManager send liquidity to the reward distributor according to each collateral type
+ */
+library LiquidationAssetManager {
+    using DecimalMath for uint256;
+
+    /**
+     * @notice Thrown when attempting to access a not registered id
+     */
+    error InvalidId(uint128 id);
+
+    /**
+     * @notice Thrown when attempting to use a wrong distributor
+     */
+    error InvalidDistributor(uint128 id, address distributor);
+
+    struct Data {
+        /**
+         * @dev Collateral Id (same as synth id)
+         */
+        uint128 id;
+        /**
+         * @dev TODO: add comment
+         */
+        address distributor;
+        /**
+         * @dev TODO: add comment
+         */
+        address[] collateralTypes;
+    }
+
+    /**
+     * @dev Load the collateral configuration data using collateral/synth id
+     */
+    function load(uint128 collateralId) internal pure returns (Data storage collateralLAMConfig) {
```

Comment:
instead of loading it here using collateralId, lets tie this to `CollateralConfiguration` so there's just one place for all collateral configuration data:
```
CollateralConguration.Data {
  LiquidationAssetManager.Data lam;
}
```